beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -428,6 +472,30 @@ impl ExecutionPayloadError {
             ExecutionPayloadError::UnverifiedNonOptimisticCandidate => false,
         }
     }
+
+    pub fn penalize_rpc_peer(&self) -> ErrorCategory {
+        // This match statement should never have a default case
+        match self {
+            // The peer has nothing to do with this error, do not penalize them.
+            ExecutionPayloadError::NoExecutionConnection => {
+                ErrorCategory::internal_non_recoverable()
+            }
+            // The peer has nothing to do with this error, do not penalize them.
+            ExecutionPayloadError::RequestFailed(_) => ErrorCategory::internal_recoverable(),
+            // Execution payload is invalid
+            ExecutionPayloadError::RejectedByExecutionEngine { .. }
```

Comment:
If EL says the payload is invalid we MUST downscore this peer right?

Reply:
Yes, the peer has already imported the block into fork choice if we are fetching it over rpc